src/pip/_internal/wheel_builder.py

```diff
@@ -273,7 +274,7 @@ def build(
     # Build the wheels.
     logger.info(
         'Building wheels for collected packages: %s',
-        ', '.join(req.name for req in requirements),
+        ', '.join(req.name for req in requirements if req.name),
```

Comment:
If we can't persuade mypy that `req.name` is not None here, I'd rather silence it than adding this `if req.name`.

Reply:
Thanks, I have gone ahead and ignored them in https://github.com/pypa/pip/pull/8337/commits/9a8bcf3f343fe692c540808b451113b147b4082a